src/snowflake/snowpark/mock/_connection.py

```diff
@@ -700,15 +703,18 @@ def execute(
             rows = [rows] if to_iter else rows
 
         # Notify query listeners.
+        self.notify_mock_query_record_listener(kwargs)
+
+        return iter(rows) if to_iter else rows
+
```

Comment:
Why do we need a new function `notify_mock_query_record_listener(kwargs)` here? Shouldn't it be sufficient to directly call ` self.notify_query_listeners` or so?

Reply:
To avoid duplicating the code relate to the mock plan query record, request-id and passing through the dataframeAst.